Handle sysctl VLANs

sgnsys3 opened the pull request:
Referring to Issue #404 

If the Interface name has a "." in the name then replace "." with "/".

konstruktoid commented:
Hi @sgnsys3 and thanks for the PR.
However it returns:
```sh
Syntax Error while loading YAML.
  found unknown escape character
```

Please test with `| regex_replace('\\.', '/')`

konstruktoid commented on tasks/sysctl.yml:
```diff
@@ -14,7 +14,7 @@
   become: true
   ansible.posix.sysctl:
-    name: net.ipv6.conf.{{ ansible_default_ipv4.interface }}.accept_ra_rtr_pref
+    name: net.ipv6.conf.{{ ansible_default_ipv4.interface | regex_replace('\.', '/') }}.accept_ra_rtr_pref
```
```console
Syntax Error while loading YAML.
  found unknown escape character
```

sgnsys3 replied:
I already test with `| regex_replace('\\.', '/')` it not working.
so that I use `| regex_replace('\.', '/')` instead.
Btw, My code already runs in my production without problem.

Could you share how to test or where an error came from? I will diagnose it again.

konstruktoid replied:
```sh
$ ansible-playbook -i 'localhost,' -c local regsys.yml 
ERROR! We were unable to read either as JSON nor YAML, these are the errors we got from each:
JSON: Expecting value: line 1 column 1 (char 0)

Syntax Error while loading YAML.
  found unknown escape character

The error appears to be in '/home/tsj/regsys.yml': line 15, column 64, but may
be elsewhere in the file depending on the exact syntax problem.

The offending line appears to be:

      ansible.builtin.debug:
        msg: "{{ ansible_default_ipv4.address | regex_replace('\.', '/') }}"
                                                               ^ here
We could be wrong, but this one looks like it might be an issue with
missing quotes. Always quote template expression brackets when they
start a value. For instance:

    with_items:
      - {{ foo }}

Should be written as:

    with_items:
      - "{{ foo }}"
$ cat regsys.yml 
---
- hosts: localhost
  gather_facts: true
  tasks:
    - name: A
      ansible.builtin.debug:
        msg: "{{ ansible_default_ipv4.address }}"
          
    - name: B
      ansible.builtin.debug:
        msg: "{{ ansible_default_ipv4.address | regex_replace('\\.', '/') }}"

    - name: C
      ansible.builtin.debug:
        msg: "{{ ansible_default_ipv4.address | regex_replace('\.', '/') }}"
$ vi regsys.yml 
$ cat regsys.yml 
---
- hosts: localhost
  gather_facts: true
  tasks:
    - name: A
      ansible.builtin.debug:
        msg: "{{ ansible_default_ipv4.address }}"
          
    - name: B
      ansible.builtin.debug:
        msg: "{{ ansible_default_ipv4.address | regex_replace('\\.', '/') }}"

#    - name: C
#      ansible.builtin.debug:
#        msg: "{{ ansible_default_ipv4.address | regex_replace('\.', '/') }}"
$ ansible-playbook -i 'localhost,' -c local regsys.yml 

PLAY [localhost] ************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************
ok: [localhost]

TASK [A] *********************************************************************************************************
ok: [localhost] => {
    "msg": "192.168.0.95"
}

TASK [B] *********************************************************************************************************
ok: [localhost] => {
    "msg": "192/168/0/95"
}

PLAY RECAP ***************************************************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
````

sgnsys3 replied:
The error will occur in case we have quoted text by double quote. 
For example
```
# No error
    - name: C
      ansible.builtin.debug:
        msg: test {{ ansible_default_ipv4.address | regex_replace('\.', '/') }}
# Error
    - name: C
      ansible.builtin.debug:
        msg: "test {{ ansible_default_ipv4.address | regex_replace('\.', '/') }}"
```
According to this yaml spec https://yaml.org/spec/1.2.2/#731-double-quoted-style
Double-quote will read escape string but single quote does not.

I will add some tests and quote sysctl name with double quote.

konstruktoid commented:
Hi again @sgnsys3 
Any updates on this one?

sgnsys3 commented:
@konstruktoid Hi Sorry for the late reply, Last month was very tight for me.
I've updated the change according to your latest code.

In Vagrantfile, I have updated the serial connection method because I have run it on Windows and it always failed
To make this easier I set it to disconnected. Or Does It need to be connected?

konstruktoid commented:
> @konstruktoid Hi Sorry for the late reply, Last month was very tight for me. I've updated the change according to your latest code.

No worries at all :)

> In Vagrantfile, I have updated the serial connection method because I have run it on Windows and it always failed To make this easier I set it to disconnected. Or Does It need to be connected?

Nah, it can be connected when needed.

konstruktoid commented:
Just one comment regarding the `Vagrantfile`

konstruktoid commented:
```diff
diff --git a/Vagrantfile b/Vagrantfile
index eb900cd..7c5c24b 100644
--- a/Vagrantfile
+++ b/Vagrantfile
@@ -22,10 +22,10 @@ Vagrant.configure("2") do |config|
       a.extra_vars = {
         "ansible_become_pass" => "vagrant",
         "ansible_python_interpreter" => "/usr/bin/python3",
-        "sshd_admin_net" => "0.0.0.0/0",
-        "sshd_allow_groups" => "vagrant sudo debian ubuntu",
-        "system_upgrade" => "no",
-        "install_aide" => "false"
+        "sshd_admin_net" => ["0.0.0.0/0"],
+        "sshd_allow_groups" => ["vagrant", "sudo", "debian", "ubuntu"],
+        "system_upgrade" => "false",
+        "install_aide" => "false",
       }
     end
   end
@@ -47,7 +47,7 @@ Vagrant.configure("2") do |config|
         "ansible_python_interpreter" => "/usr/bin/python3",
         "sshd_admin_net" => ["0.0.0.0/0"],
         "sshd_allow_groups" => ["vagrant", "sudo", "debian", "ubuntu"],
-        "system_upgrade" => "no",
+        "system_upgrade" => "false",
      }
     end
   end
```

konstruktoid commented:
Closes https://github.com/konstruktoid/ansible-role-hardening/issues/404

sgnsys3 commented:
@konstruktoid Updated

konstruktoid commented:
Thanks @sgnsys3!
